engine/crates/composition/src/composition_ir.rs

```diff
@@ -129,7 +129,27 @@ impl CompositionIr {
         id
     }
 
-    pub(crate) fn insert_enum_value(&mut self, enum_id: federated::EnumId, value: StringWalker<'_>) {
+    pub(crate) fn insert_enum_value(
+        &mut self,
+        enum_id: federated::EnumId,
+        value: StringWalker<'_>,
+        deprecation: Option<Option<StringWalker<'_>>>,
+    ) {
+        let mut composed_directives = Vec::new();
+
+        if let Some(deprecation) = deprecation {
+            let arguments = match deprecation {
+                Some(reason) => vec![(
+                    self.insert_static_str("reason"),
+                    federated::Value::String(self.insert_string(reason)),
+                )],
+                None => Vec::new(),
+            };
+            let name = self.insert_static_str("deprecated");
```

Comment:
There is a bit of duplication here. We'll probably want a module for directive composition that handles the shared logic.